File: frontend/src/microservices/operations/objects/UpdateAirlineManagerProfilePlannerRequest.ts
export type UpdateAirlineManagerProfilePlannerRequest = {
  managerId: string
  displayName: string
  airlineName: string
  airlineCode: string
  logoAssetPath?: string | null
}

export const updateAirlineManagerProfilePlannerRequestFromJson = (json: string): UpdateAirlineManagerProfilePlannerRequest =>
  JSON.parse(json) as UpdateAirlineManagerProfilePlannerRequest

export const updateAirlineManagerProfilePlannerRequestToJson = (value: UpdateAirlineManagerProfilePlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/airline/objects/UpdateAirlineManagerProfilePlannerRequest.scala
// 这个文件只定义 airline 管理员资料更新请求。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UpdateAirlineManagerProfilePlannerRequest(
    managerId: String,
    displayName: String,
    airlineName: String,
    airlineCode: String,
    logoAssetPath: Option[String]
)
object UpdateAirlineManagerProfilePlannerRequest:
  given sourceEncoder: Encoder[UpdateAirlineManagerProfilePlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[UpdateAirlineManagerProfilePlannerRequest] = deriveDecoder
